### PR TITLE
libbsd: Update to 0.8.7

### PR DESCRIPTION
**Pull request by diizzyy**
Update libbsd to 0.8.7
Remove glibc dependency
Clean up InstallDev and install entries
Use /usr path for consistency
Cherry pick patches from upstream to fix musl compilation

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

### CODE REVIEW

**Comment by diizzyy** [comment]
@pprindeville 
Can you have a look at this? Perl should also be affected.

**Comment by diizzyy** [comment]
Compile tested: mvebu, Linksys WRT3200ACM, OpenWrt Master
Runtime tested: mvebu, Linksys WRT3200ACM, OpenWrt Master

File: package/libs/libbsd/Makefile [hunk 1]
```diff
@@ -27,30 +26,17 @@ define Package/libbsd/description
...
-	$(CP) \
-		$(PKG_INSTALL_DIR)/lib/libbsd.so* \
-		$(1)/lib/
-
-	$(CP) \
-		$(PKG_INSTALL_DIR)/usr/include/* \
-		$(1)/usr/include/
-
-	( cd $(1)/lib ; $(LN) libbsd.so.$(PKG_VERSION) libbsd.so )
+	$(INSTALL_DIR) $(1)/usr/lib
```
Comment (pprindeville) [inline package/libs/libbsd/Makefile]: Can all 3 directories be created on the same line?

**Review by pprindeville — CHANGES_REQUESTED** [review]
Minor change requested...  Also, what are the reverse dependencies of libbsd?

File: package/libs/libbsd/Makefile (hunk 1 above)
Comment (diizzyy) [reply package/libs/libbsd/Makefile]: Dunno? :-)
You like ```$(INSTALL_DIR) $(1)/usr/lib $(INSTALL_DIR) $(1)/usr/lib/pkgconfig $(INSTALL_DIR) $(1)/usr/include``` ?
Wouldn't that get too long?

**Comment by diizzyy** [comment]
libbsd is used (mentioned in some way) in lldpd, libtalloc, darkstat, perl and hfsprogs.
I was going to use it for a new package but as it turned out it wasn't needed but at that time I had already bumped libbsd.

**Comment by neheb** [comment]
Seems outdated.

Also, some of these dependencies seem bogus. lldpd at least does not require it. libtalloc does not either.

I'd rather patch out the packages that require libbsd.

File: package/libs/libbsd/Makefile (hunk 1 above)
Comment (pprindeville) [reply package/libs/libbsd/Makefile]: Should be fine

**Comment by diizzyy** [comment]
@neheb 
Yes, 0.9.1 is out but it's also been sitting for over a month in the pipeline....

lidpd uses it in some ways apparently
https://github.com/vincentbernat/lldpd/blob/master/configure.ac#L175

As for libtalloc (can probably be removed)
```
In ./lib/replace/wscript
"    # libbsd on some platforms provides strlcpy and strlcat"
```

As I'm not the maintainer of those packages I didn't look too much into it.

File: package/libs/libbsd/Makefile (hunk 1 above)
Comment (diizzyy) [reply package/libs/libbsd/Makefile]: I honestly think that's harder to read but if the preferred way I can change it.
If we should keep the style it needs to be broken down into two lines which doesn't save much in the end?

Comment (pprindeville) [reply package/libs/libbsd/Makefile]: It’s about seeing up front all the directories that need to be created without having to search them out.

**Comment by hauke** [comment]
Thank you for the pull request, it was applied to master.

**Pull request closed without merging**